sdk/communication/azure-communication-networktraversal/src/main/java/com/azure/communication/networktraversal/CommunicationRelayClientBuilder.java

```diff
@@ -0,0 +1,303 @@
+     * @return The updated {@link CommunicationRelayClientBuilder} object.
+     */
+    public CommunicationRelayClientBuilder clientOptions(ClientOptions clientOptions) {
+        this.clientOptions = clientOptions;
+        return this;
+    }
+
+    /**
+     * Sets the configuration object used to retrieve environment configuration values during building of the client.
+     *
+     * @param configuration Configuration store used to retrieve environment configurations.
+     * @return the updated CommunicationRelayClientBuilder object
+     */
+    public CommunicationRelayClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpLogOptions} for service requests.
+     *
+     * @param logOptions The logging configuration to use when sending and receiving HTTP requests/responses.
+     * @return the updated CommunicationRelayClientBuilder object
+     */
+    public CommunicationRelayClientBuilder httpLogOptions(HttpLogOptions logOptions) {
+        this.httpLogOptions = logOptions;
+        return this;
+    }
+
+    /**
+     * Sets the {@link RetryPolicy} that is used when each request is sent.
+     *
+     * @param retryPolicy User's retry policy applied to each request.
+     * @return The updated {@link CommunicationRelayClientBuilder} object.
+     */
+    public CommunicationRelayClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        this.retryPolicy = retryPolicy;
+        return this;
+    }
+
+    /**
+     * Sets the {@link CommunicationRelayServiceVersion} that is used when making API requests.
+     * <p>
+     * If a service version is not provided, the service version that will be used will be the latest known service
+     * version based on the version of the client library being used. If no service version is specified, updating to a
+     * newer version of the client library will have the result of potentially moving to a newer service version.
+     * <p>
+     * Targeting a specific service version may also mean that the service will return an error for newer APIs.
+     *
+     * @param version {@link CommunicationRelayServiceVersion} of the service to be used when making requests.
+     * @return the updated CommunicationRelayClientBuilder object
+     */
+    public CommunicationRelayClientBuilder serviceVersion(CommunicationRelayServiceVersion version) {
+        return this;
+    }
+
+    /**
+     * Create asynchronous client applying HMACAuthenticationPolicy, UserAgentPolicy,
+     * RetryPolicy, and CookiePolicy.
+     * Additional HttpPolicies specified by additionalPolicies will be applied after them
+     *
+     * @return CommunicationRelayAsyncClient instance
+     */
+    public CommunicationRelayAsyncClient buildAsyncClient() {
+        Objects.requireNonNull(endpoint, "'ednpoint' cannot be null.");
+        Objects.requireNonNull(customPolicies, "'customPolicies' cannot be null.");
```

Comment:
`customPolicies` can be null. If it is null, no additional policies are added to the pipeline.